app/sprinkles/account/src/Authenticate/PasswordSecurity.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * UserFrosting (http://www.userfrosting.com)
+ *
+ * @link      https://github.com/userfrosting/UserFrosting
+ * @copyright Copyright (c) 2019 Alexander Weissman
+ * @license   https://github.com/userfrosting/UserFrosting/blob/master/LICENSE.md (MIT License)
+ */
+
+namespace UserFrosting\Sprinkle\Account\Authenticate;
+
+use Illuminate\Cache\Repository as Cache;
+use UserFrosting\Support\Repository\Repository as Config;
+
+/**
+ * Handles enhanced password security methods for integration with Have I Been Pwned.
+ *
+ * @see https://haveibeenpwned.com/API/v2
+ * @author Amos Folz
+ */
+class PasswordSecurity
+{
+    /**
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * @var Cache
+     */
+    protected $cache;
+
+    /**
+     * Create a new PasswordSecurity object.
+     *
+     * @param Cache  $cache  Cache service instance
+     * @param Config $config Config object that contains security settings.
+     */
+    public function __construct(Cache $cache, Config $config)
+    {
+        $this->cache = $cache;
+        $this->config = $config;
+    }
+
+    /**
+     * Generates a list of compromised passwords stored in cache or by querying Have I Been Pwned API.
+     *
+     * First check the cache to see if the hash prefix is stored.
+     * If not found in cache, query Have I Been Pwned API and store response in cache.
+     * @param  string $password
+     * @return string A numeric string representing the number of times a password has been compromised.
```

Comment:
A string seems odd here. Representing how?

Reply:
Changed to `int` with new comment